### PR TITLE
Fixes #9321 - Add option to prevent option domain from being set in dhcp...

### PR DESCRIPTION
**Pull request by Moncky**
...d.conf

### CODE REVIEW

**Comment by ekohl** [comment]
I'd like to get the design straight in https://github.com/theforeman/puppet-dhcp/pull/38 before discussing this particular PR (while keeping it in mind of course).

**Comment by ekohl** [comment]
Now the puppet-dhcp part is merged, would you mind updating this PR?

**Comment by Moncky** [comment]
I have updated this so that by default it populates the array, I think this is something you would wan't to explicitly unset.  I expect these tests will fail and will have a look at them when they do.

File: manifests/params.pp [hunk 1]
```diff
@@ -112,12 +112,13 @@
...
-  $dhcp_interface   = 'eth0'
-  $dhcp_gateway     = '192.168.100.1'
-  $dhcp_range       = false
+  $dhcp                   = false
+  $dhcp_listen_on         = 'https'
+  $dhcp_managed           = true
+  $dhcp_interface         = 'eth0'
+  $dhcp_gateway           = '192.168.100.1'
+  $dhcp_range             = false
+  $dhcp_option_domain     = [${::domain}]
```
Comment (ekohl) [inline manifests/params.pp]: No need to use braces, just ```[$::domain]``` should suffice.

**Comment by Moncky** [comment]
Ahh of course, updated that now.  Thanks for the pointer

File: manifests/init.pp [hunk 2]
```diff
@@ -152,6 +152,8 @@
 #
 # $dhcp_managed::               DHCP is managed by Foreman proxy
 #                               type:boolean
+# $dhcp_use_option_domain::     DHCP use the dhcpd config option domain-name
+#                               type:boolean
```
Comment (ekohl) [inline manifests/init.pp]: This is now an array.

**Comment by ekohl** [comment]
You should also pass it [here](https://github.com/theforeman/puppet-foreman_proxy/blob/468eb0c1ed5695339f6256330282b4bb89775d8e/manifests/proxydhcp.pp#L28)

**Comment by Moncky** [comment]
The last commit there prevents the array created in the param.pp from being put into another array, which caused the compilation of the erb template fail.

**Comment by ekohl** [comment]
I wonder how easy it is to use the installer and input an empty array, but your use case is direct use of the modules?

@domcleal I think the current approach is the right one, but maybe you have another opinion?

**Comment by domcleal** [comment]
I think it's the right approach too, but I don't know about the installer/kafo.  I can't seem to reset an array with `--foreman-proxy-trusted-hosts=`, so maybe that's a kafo "feature" to request (@ares, agreed?).

**Comment by Moncky** [comment]
Yes, I'm using the modules directly.  I used the foreman-installer to boot strap a basic system then have over-ridden various options in using the smart class parameter.

**Comment by ekohl** [comment]
@Moncky  I did the same thing and IMHO a very important use case of the modules :)

**Comment by ekohl** [comment]
Let's just merge this. I think it's fine if this particular use case isn't easily supported in the installer. @domcleal @mmoll?

**Comment by domcleal** [comment]
Agreed, thanks @Moncky!

I filed http://projects.theforeman.org/issues/9509 to figure out the kafo support.

**Pull request closed without merging**